pkg/parser/parser_test.go

```diff
@@ -0,0 +1,40 @@
+package parser
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/libopenstorage/cloudops"
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	testYamlFilePath = "/tmp/cloudops-test.yaml"
+)
+
+func TestStorageDecisionMatrixParser(t *testing.T) {
+	inputMatrix := cloudops.StorageDecisionMatrix{
+		Rows: []cloudops.StorageDecisionMatrixRow{
+			cloudops.StorageDecisionMatrixRow{
+				IOPS:         uint32(1000),
+				MinSize:      uint64(100),
+				MaxSize:      uint64(200),
+				InstanceType: "foo",
+			},
+			cloudops.StorageDecisionMatrixRow{
+				IOPS:         uint32(2000),
+				MinSize:      uint64(200),
+				MaxSize:      uint64(400),
+				InstanceType: "bar",
+			},
+		},
+	}
+	p := NewStorageDecisionMatrixParser()
+	err := p.MarshalToYaml(&inputMatrix, testYamlFilePath)
```

Comment:
this test doesn't validate if this works with a given yaml file since we are marshalling the struct into yaml and then unmarshalling the same file.

We should rather be starting the test by calling `UnmarshalFromYaml` on https://github.com/libopenstorage/cloudops/blob/master/azure/testspecs/azure-storage-decision-matrix.yaml.

Reply:
done!

Found an issue with the azure test spec after adding the test :)